### ISSUE TITLE
some error on fcoin

### ISSUE THREAD

**Issue by shaohaojiecoder** [ccxt/ccxt#3331]
ATTENTION!!!

MUST READ THIS BEFORE SUBMITTING ISSUES:

https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-submit-an-issue

- OS: windows
- Programming Language version:python
- CCXT version:1.15.19
- Exchange:fcoin
- Method:fetch_order

when you fetch an order  , it returns 
############################################
 Traceback (most recent call last):
  File "C:\Users\shaoh\AppData\Local\Programs\Python\Python36\Lib\site-packages\ccljmltrader\test.py", line 10, in <module>
    print(ex.fetch_order(ret))
  File "C:\Users\shaoh\AppData\Local\Programs\Python\Python36\lib\site-packages\ccxt\fcoin.py", line 409, in fetch_order
    return self.parse_order(response)
  File "C:\Users\shaoh\AppData\Local\Programs\Python\Python36\lib\site-packages\ccxt\fcoin.py", line 354, in parse_order
    id = order['id']
KeyError: 'id'

**Comment by kroitor** [comment]
@shaohaojiecoder need your verbose request and response, as described here: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-submit-an-issue

**Comment by shaohaojiecoder** [comment]
the order id is ''5e2Wvd_maPFeupeNzGjha2YNfDZETidiTU3Ah2dvJeM=''

**Comment by kroitor** [comment]
@shaohaojiecoder you're doing everything exactly the opposite.

1. I told you to go read that link and to post your verbose request and response. You didn't.
2. I told you to not post your apiKey and secret – you posted them.
3. I told you that we need your verbose request and response, instead you posted just the code snippet (thx, but it's not enough).

You need to pay attention, otherwise it will take years to actually help you resolve this... I hope you don't want this to take that much.

**Comment by shaohaojiecoder** [comment]
so sorry, i have pasted the verbose requstes and response.
I post an order to the interface fetch_order
but it returns  an error

**Comment by shaohaojiecoder** [comment]
Am I clear?@kroitor

**Comment by kroitor** [comment]
> i have pasted the verbose requstes and response.

No, you haven't.

> Am I clear?@kroitor

```Python
import ccxt
ex = ccxt.fcoin({
    'verbose': True,  # ←----------------------- add this and copy the output here
})
print(ex.fetch_order('5e2Wvd_maPFeupeNz342fDZETidiTU3Ah2dvJeM='))
```

Can you copy the entire output (it contains the verbose request and response) here?

**Comment by shaohaojiecoder** [comment]
Request: GET https://api.fcoin.com/v2/public/symbols {'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.71 Safari/537.36', 'Accept-Encoding': 'gzip, deflate'} None

Response: GET https://api.fcoin.com/v2/public/symbols 200 {'Date': 'Fri, 06 Jul 2018 09:55:25 GMT', 'Content-Type': 'application/json;charset=UTF-8', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'Server': 'nginx'} {"status":0,"data":[{"name":"btcusdt","base_currency":"btc","quote_currency":"usdt","price_decimal":2,"amount_decimal":4},{"name":"ethusdt","base_currency":"eth","quote_currency":"usdt","price_decimal":2,"amount_decimal":4},{"name":"bchusdt","base_currency":"bch","quote_currency":"usdt","price_decimal":2,"amount_decimal":4},{"name":"ltcusdt","base_currency":"ltc","quote_currency":"usdt","price_decimal":2,"amount_decimal":4},{"name":"ftusdt","base_currency":"ft","quote_currency":"usdt","price_decimal":6,"amount_decimal":2},{"name":"fteth","base_currency":"ft","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"zipeth","base_currency":"zip","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"etcusdt","base_currency":"etc","quote_currency":"usdt","price_decimal":2,"amount_decimal":4},{"name":"ftbtc","base_currency":"ft","quote_currency":"btc","price_decimal":8,"amount_decimal":2},{"name":"omgeth","base_currency":"omg","quote_currency":"eth","price_decimal":6,"amount_decimal":4},{"name":"btmusdt","base_currency":"btm","quote_currency":"usdt","price_decimal":4,"amount_decimal":2},{"name":"zrxeth","base_currency":"zrx","quote_currency":"eth","price_decimal":6,"amount_decimal":2},{"name":"bnbusdt","base_currency":"bnb","quote_currency":"usdt","price_decimal":4,"amount_decimal":2},{"name":"zipusdt","base_currency":"zip","quote_currency":"usdt","price_decimal":6,"amount_decimal":2},{"name":"xrpusdt","base_currency":"xrp","quote_currency":"usdt","price_decimal":4,"amount_decimal":2},{"name":"fieth","base_currency":"fi","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"fiusdt","base_currency":"fi","quote_currency":"usdt","price_decimal":6,"amount_decimal":2},{"name":"icxeth","base_currency":"icx","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"zileth","base_currency":"zil","quote_currency":"eth","price_decimal":6,"amount_decimal":4},{"name":"aeeth","base_currency":"ae","quote_currency":"eth","price_decimal":6,"amount_decimal":2},{"name":"777eth","base_currency":"777","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"guseth","base_currency":"gus","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"cccxeth","base_currency":"cccx","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"bancaeth","base_currency":"banca","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"praeth","base_currency":"pra","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"dcceth","base_currency":"dcc","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"ssseth","base_currency":"sss","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"mdteth","base_currency":"mdt","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"tsteth","base_currency":"tst","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"pmdeth","base_currency":"pmd","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"rteeth","base_currency":"rte","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"xpseth","base_currency":"xps","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"tcteth","base_currency":"tct","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"dwseth","base_currency":"dws","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"ngoteth","base_currency":"ngot","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"ateth","base_currency":"at","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"soceth","base_currency":"soc","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"blzeth","base_currency":"blz","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"ocneth","base_currency":"ocn","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"datxeth","base_currency":"datx","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"gtceth","base_currency":"gtc","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"leteth","base_currency":"let","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"dageth","base_currency":"dag","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"yeeeth","base_currency":"yee","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"aaaeth","base_currency":"aaa","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"nceth","base_currency":"nc","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"aiteth","base_currency":"ait","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"arpeth","base_currency":"arp","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"grameth","base_currency":"gram","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"ifoodeth","base_currency":"ifood","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"hpceth","base_currency":"hpc","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"sgcceth","base_currency":"sgcc","quote_currency":"eth","price_decimal":8,"amount_decimal":2},{"name":"3dbeth","base_currency":"3db","quote_currency":"eth","price_decimal":8,"amount_decimal":2}]}

Request: GET https://api.fcoin.com/v2/orders/AZa2WEeCWpbxzPrBEV5oUoYiNaU-GhEIfaS_RUICAI8= {'FC-ACCESS-KEY': 'c866783b4579476d9d6790ad16f2bd2f', 'FC-ACCESS-SIGNATURE': 'xPM0uUkBz0SI72BPHhYVztMOd5k=', 'FC-ACCESS-TIMESTAMP': '1530870925427', 'Content-Type': 'application/json', 'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.71 Safari/537.36', 'Accept-Encoding': 'gzip, deflate'} None

Response: GET https://api.fcoin.com/v2/orders/AZa2WEeCWpbxzPrBEV5oUoYiNaU-GhEIfaS_RUICAI8= 200 {'Date': 'Fri, 06 Jul 2018 09:55:25 GMT', 'Content-Type': 'application/json;charset=UTF-8', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'Server': 'nginx'} {"status":0,"data":{"id":"AZa2WEeCWpbxzPrBEV5oUoYiNaU-GhEIfaS_RUICAI8=","symbol":"arpeth","amount":"200.000000000000000000","price":"0.000200000000000000","created_at":1530869462702,"type":"limit","side":"buy","filled_amount":"0.000000000000000000","executed_value":"0.000000000000000000","fill_fees":"0.000000000000000000","source":"api","state":"submitted"}}
Traceback (most recent call last):
  File "C:\Users\shaoh\AppData\Local\Programs\Python\Python36\Lib\site-packages\ccljmltrader\test.py", line 10, in <module>
    print(ex.fetch_order(ret))
  File "C:\Users\shaoh\AppData\Local\Programs\Python\Python36\lib\site-packages\ccxt\fcoin.py", line 409, in fetch_order
    return self.parse_order(response)
  File "C:\Users\shaoh\AppData\Local\Programs\Python\Python36\lib\site-packages\ccxt\fcoin.py", line 354, in parse_order
    id = order['id']
KeyError: 'id'
[Finished in 6.4s with exit code 1]

**Comment by kroitor** [comment]
@shaohaojiecoder ok, much better now, thx! This was a bug, I've added a fix for it, and it is available in the new version 1.15.21. So if you just run `pip install ccxt --upgrade`, your code snippet should work fine from there.

```Python
import ccxt
print('CCXT Version:', ccxt.__version__)  # to verify that your version is updated
ex = ccxt.fcoin({
    'enableRateLimit': True,  # ←----------------------- highly recommended to add this
})

# this should work normally:
print(ex.fetch_order('5e2Wvd_maPFeupeNz342fDZETidiTU3Ah2dvJeM='))
```

Also, I'd recommend to read the entire Manual from the beginning to the end – you'll find a lot of useful info and it will save lots of debugging time for you as well. Because from your current code it doesn't seem that you're using rate limiting, whereas you should've enabled it.

The Manual itself is here:

- https://github.com/kroitor/ccxt/wiki
- https://github.com/kroitor/ccxt/wiki/Manual

Hope this helps.

**Comment by shaohaojiecoder** [comment]
thanks @kroitor  /smail
